docs/src/content/docs/guides/components.mdx

```diff
@@ -195,3 +195,31 @@ The code above generates the following on the page:
 A list of all available icons is shown below with their associated names. Click an icon to copy the component code for it.
 
 <IconsList />
+
+### Code
+
+Starlight uses [Expressive Code](https://expressive-code.com/) to render code blocks with extended formatting possibilities. In addition to fenced code blocks in markdown & MDX documents, Expressive Code can display syntax-highlighted code snippets using its [`<Code>` component](https://expressive-code.com/key-features/code-component/).
+
+The `<Code>` component provides props like `code`, `lang` or `meta` that allow you to dynamically define a code block's contents. Using this component, you can render code blocks from variables or data coming from external sources like files, databases or APIs.
+
+```mdx
+# src/content/docs/example.mdx
+
+import { Code } from '@astrojs/starlight/components';
+
+export const exampleCode = `console.log('This could come from a file or CMS!');`;
+export const fileName = 'example.js';
+export const highlights = ["file", "CMS"];
+
+<Code code={exampleCode} lang="js" title={fileName} mark={highlights} />
+```
+
+The code above generates the following on the page:
+
+import { Code } from '@astrojs/starlight/components';
+
+export const exampleCode = `console.log('This could come from a file or CMS!');`;
+export const fileName = 'example.js';
+export const highlights = ["file", "CMS"];
+
+<Code code={exampleCode} lang="js" title={fileName} mark={highlights} />
```

Comment:
Since it came up several times recently, what do you think of adding something like this to explicitly document the import version of usage?

````suggestion
<Code code={exampleCode} lang="js" title={fileName} mark={highlights} />

#### Code transclusion

Use [Vite’s `?raw` import suffix](https://vitejs.dev/guide/assets#importing-asset-as-string) to import any code file as a string.
You can then pass this imported string to the `<Code>` component to include it on your page.

```mdx title="src/content/docs/example.mdx" "?raw"
import { Code } from '@astrojs/starlight/components';
import importedCode from '/src/env.d.ts?raw';

<Code code={importedCode} lang="ts" title="src/env.d.ts" />
```

The code above generates the following on the page:

import importedCode from '/src/env.d.ts?raw';

<Code code={importedCode} lang="ts" title="src/env.d.ts" />
````

Reply:
Love it! Great idea to add this example. Btw, this is the first time I've read the word "transclusion", I'm just trusting you that this is the correct term. :)

Reply:
Haha, I honestly only learned it after working on Starlight and seeing it used by technical writers and some other tools. It’s a bit specialised (so I might double check with Sarah if she thinks there’s a better way to say it), but basically a technical term meaning including other content by reference in a page ([Wikipedia](https://en.wikipedia.org/wiki/Transclusion)).